app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -383,4 +388,28 @@ public function getPrefetchCacheIdentifier($quote, $addressData)
 
         return md5($cacheIdentifier);
     }
+
+    /**
+     * @return mixed
+     */
+    protected function isPOBoxAllowed()
+    {
+        return Mage::getStoreConfig('payment/boltpay/allow_po_box');
+    }
+
+    /**
+     * @param $address1
+     * @param $address2
+     * @return mixed
+     */
+    protected function doesAddressContainPOBox($address1, $address2)
```

Comment:
@johnmccombs1 This is a good candidate for testing. Can you write a unit test?

Reply:
same input proposal (please add more)

match
 - P.O. Box 66
 - Post Box 123
 - Post Office Box 456

not match
 - Post street
 - 2 Box

Reply:
These checks have been added along with a few more.